inlang/source-code/sdk/src/resolve-modules/plugins/resolvePlugins.ts

```diff
@@ -116,12 +117,23 @@ export const resolvePlugins: ResolvePluginsFunction = async (args) => {
 	}
 
 	// --- LOADMESSAGE / SAVEMESSAGE NOT DEFINED ---
+
 	if (
-		typeof result.data.loadMessages !== "function" ||
-		typeof result.data.saveMessages !== "function"
+		!experimentalPersistence &&
+		(typeof result.data.loadMessages !== "function" ||
+			typeof result.data.saveMessages !== "function")
 	) {
 		result.errors.push(new PluginsDoNotProvideLoadOrSaveMessagesError())
```

Comment:
can be turned into an if else statement

if (experimentalPersistence){

} else if()